rsyslog/undefined_field/undefined_field.go

```diff
@@ -0,0 +1,392 @@
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			rawStr := scanner.Text()
+			// skip comments
+			if strings.HasPrefix(rawStr, "#") {
+				continue
+			}
+			configMap := make(map[string]interface{})
+			if err := json.Unmarshal([]byte(rawStr), &configMap); err != nil {
+				fmt.Fprintln(logfile, "ERROR: Failed to parse config file [%s]: [%v]", undefined_config, err)
+				panic(fmt.Errorf("Failed to parse config file [%s]: [%v]", undefined_config, err))
+			}
+
+			if val, ok := getMapStringValue(configMap, "UNDEFINED_DEBUG"); ok {
+				if rval, err := strconv.ParseBool(val); err == nil {
+					undefined_debug = rval
+				} else {
+					fmt.Fprintln(logfile, "ERROR: Invalid UNDEFINED_DEBUG value [%s]: [%v]", val, err)
+					panic(fmt.Errorf("Invalid UNDEFINED_DEBUG value [%s]: [%v]", val, err))
+				}
+			}
+			if val, ok := getMapStringValue(configMap, "MERGE_JSON_LOG"); ok {
+				if rval, err := strconv.ParseBool(val); err == nil {
+					merge_json_log = rval
+				} else {
+					fmt.Fprintln(logfile, "ERROR: Invalid MERGE_JSON_LOG value [%s]: [%v]", val, err)
+					panic(fmt.Errorf("Invalid MERGE_JSON_LOG value [%s]: [%v]", val, err))
+				}
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_USE_UNDEFINED"); ok {
+				if rval, err := strconv.ParseBool(val); err == nil {
+					use_undefined = rval
+				} else {
+					fmt.Fprintln(logfile, "ERROR: Invalid CDM_USE_UNDEFINED value [%s]: [%v]", val, err)
+					panic(fmt.Errorf("Invalid CDM_USE_UNDEFINED value [%s]: [%v]", val, err))
+				}
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_DEFAULT_KEEP_FIELDS"); ok {
+				default_keep_fields = val
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_EXTRA_KEEP_FIELDS"); ok {
+				extra_keep_fields = val
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_UNDEFINED_NAME"); ok {
+				undefined_name = val
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_KEEP_EMPTY_FIELDS"); ok {
+				tmp_keep_empty_fields = val
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_UNDEFINED_TO_STRING"); ok {
+				if rval, err := strconv.ParseBool(val); err == nil {
+					undefined_to_string = rval
+				} else {
+					fmt.Fprintln(logfile, "ERROR: Invalid CDM_UNDEFINED_TO_STRING value [%s]: [%v]", val, err)
+					panic(fmt.Errorf("Invalid CDM_UNDEFINED_TO_STRING value [%s]: [%v]", val, err))
+				}
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_UNDEFINED_DOT_REPLACE_CHAR"); ok {
+				undefined_dot_replace_char = val
+			}
+			if val, ok := getMapStringValue(configMap, "CDM_UNDEFINED_MAX_NUM_FIELDS"); ok {
+				var rval64 int64
+				if rval64, err = strconv.ParseInt(val, 10, 64); err == nil {
+					undefined_max_num_fields = rval64
+				} else {
+					fmt.Fprintln(logfile, "ERROR: Invalid CDM_UNDEFINED_MAX_NUM_FIELDS value [%s]: [%v]", val, err)
+					panic(fmt.Errorf("Invalid CDM_UNDEFINED_MAX_NUM_FIELDS value [%s]: [%v]", val, err))
+				}
+			}
+		}
+		defer config.Close()
```

Comment:
this should be the first statement after the `open`

Reply:
:) all right.